tox.ini

```diff
@@ -48,3 +48,19 @@ allowlist_externals =
     just
 commands =
     just run-jupyter {posargs}
+
+[mypy]
+python_version = 3.12
+files = syftbox/
+disable_error_code = attr-defined, valid-type, no-untyped-call, arg-type
```

Comment:
@khoaguin We’re being pretty relaxed with the strictness here, which could let through some actual errors that should be addressed.

For example, in `syftbox/server/db/file_store.py#172-173`:
```python
metadata = hash_file(abs_path, root_dir=self.server_settings.snapshot_folder)
db.save_file_metadata(cursor, metadata)
```

Here, `metadata` is of type `Optional[FileMetadata]`, but `db.save_file_metadata` expects a non-optional `FileMetadata`. This isn’t flagged by mypy because of the `arg-type` ignore rule. Ideally, we would have added a `None` check in this case.

Are we planning to enable mypy in phases, starting with relaxed rules and then tightening them up gradually?

Reply:
Thanks for the review, Tauquir. Yes indeed I put these rules to ignore some error that require to many logical changes to the code and we can gradually remove the `disable_error_code`. I almost forgot this, thank you!